drivers/video/video_stm32_dcmi.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright (c) 2024 Charles Dias <charlesdias.cd@outlook.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_dcmi
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/drivers/dma/dma_stm32.h>
+
+#include <stm32_ll_dma.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(video_stm32_dcmi, CONFIG_STM32_DCMI_LOG_LEVEL);
+
+#ifndef CONFIG_SOC_SERIES_STM32H7X
+	#error "This driver version is only supported on STM32H7X series"
+#endif
```

Comment:
This is implicit from dts definition. I prefer we avoid these checks which end up being hard to maintain and counter productive in the long term.

Reply:
Removed.